tests/_commands/test_export_onnx.py

```diff
@@ -0,0 +1,157 @@
+    """Fixture that creates an ONNX model file for testing."""
+    onnx_path: Path = tmp_path / "model.onnx"
+    export_onnx.export_onnx(
+        out=onnx_path,
+        checkpoint=dummy_checkpoint_path,
+    )
+    return onnx_path
+
+
+def test_export_succeeds(tmp_path: Path, dummy_checkpoint_path: Path) -> None:
+    """Test that ONNX export succeeds and creates a valid file."""
+    onnx_path: Path = tmp_path / "model.onnx"
+
+    # Test export function
+    export_onnx.export_onnx(
+        out=onnx_path,
+        checkpoint=dummy_checkpoint_path,
+    )
+
+    # Verify file exists and has content
+    assert onnx_path.exists(), "ONNX file should be created"
+    assert onnx_path.stat().st_size > 0, "ONNX file should not be empty"
+
+
+def test_export_with_nonexistent_weights(tmp_path: Path) -> None:
+    """Test that export fails gracefully with nonexistent weights file."""
+    onnx_path: Path = tmp_path / "model.onnx"
+    nonexistent_weights = tmp_path / "nonexistent_weights.pth"
+
+    with pytest.raises((FileNotFoundError, RuntimeError)):
+        export_onnx.export_onnx(
+            out=onnx_path,
+            checkpoint=nonexistent_weights,
+        )
+
+
+@pytest.mark.skipif(
+    sys.version_info < (3, 10), reason="Requires Python 3.10 or higher for typing."
+)
+def test_export_parameters() -> None:
+    """Test that export function and configs have the same parameters and default values."""
+    helpers.assert_same_params(a=ExportONNXConfig, b=export_onnx.export_onnx)
+
+
+def test_onnx_model_consistency_check(onnx_model_path: Path) -> None:
+    """Test that the exported ONNX model passes ONNX validation."""
+    onnx_model: onnx.ModelProto = onnx.load(str(onnx_model_path))
+
+    try:
+        onnx.checker.check_model(onnx_model, full_check=True)
+    except Exception as e:
+        pytest.fail(f"ONNX model validation failed: {e}")
+
+
+def test_shape_inference_runs(onnx_model_path: Path) -> None:
+    """Test that shape inference runs without errors."""
+    model: onnx.ModelProto = onnx.load(str(onnx_model_path))
+
+    try:
+        inferred: onnx.ModelProto = onnx.shape_inference.infer_shapes(model)
+    except Exception as e:
+        pytest.fail(f"Shape inference failed: {e}")
+
+    # Check that inference produced results
+    assert inferred.graph, "Inferred model should have a graph"
+    assert len(inferred.graph.input) == 1, "Inferred model should have one input"
+    assert len(inferred.graph.output) == 2, "Inferred model should have two outputs"
+
+    # Verify input has shape information
+    input_shape = inferred.graph.input[0].type.tensor_type.shape
+    assert input_shape.dim, "Input should have shape dimensions"
+
+
+@pytest.mark.parametrize("batch_size", [1, 4, 8])
+def test_export_dynamic_batch(onnx_model_path: Path, batch_size: int) -> None:
+    """Test that the exported model works with different batch sizes."""
+    # Create input with the specified batch size
```

Comment:
Would drop this test. Instead the main `test_export_onnx` test should test the resulting model with a dummy input.